README.md

```diff
@@ -1,134 +1,121 @@
 
-There are 7 steps to configure the `SmartSelect`:
+There are 7 steps to configure a `SmartSelect`:
 
-1. Import the SmartSelect and SmartSelect.Utilities module and expose the `SearchUnion` type. We will get back to why we need the `SearchUnion` later.
+1. Import the select and add it to your model providing the datatype of the data to be selected.
 
 ```elm
-import SmartSelectSingle as SingleSelect
-import SmartSelect.Utilities exposing (SearchUnion(..))
-```
-
-2. Add the smart select to the model providing the datatype of the data to be selected and initialize it in the model init
+import SingleSelectLocal as Select
 
-```elm
 type alias Product =
     { id : Int
     , name : String
     , price : String
     }
 
 type alias Model =
-    { ...
-    , singleSelect : SingleSelect.SmartSelect Product
+    { products : List Product
+    , select : Select.SmartSelect Msg Product
     }
-
-init : ( Model, Cmd Msg )
-init =
-    ( { ...
-      , singleSelect = SingleSelect.init
-      }
-    )
 ```
 
-3. One message needs to be defined indicating that an update to the smart select has been triggered and needs to be handled. This message expects a tuple containing a `SingleSelect.Msg` and the selection state. **_Note:_** The selection state for the `SmartSelectMulti` is a List of selected entities as opposed to a Maybe entity for `SmartSelectSingle`.
+2. Define a `Msg` to handle updates from the select.
 
 ```elm
 type Msg
     = ...
-    | HandleSelectUpdate ( SingleSelect.Msg Product, Maybe Product )
+    | HandleSelectUpdate (Select.Msg Product)
 ```
 
-4. Configure the settings for the smart select. The `Settings` type as defined in the `SmartSelectSingle` module is below:
+3. Configure the `Settings` for the select. The `internalMsg` field takes a function that expects a `Select.Msg` and returns the `Msg` we defined in step 2. Please refer to the module documentation for more information on the settings.
 
 ```elm
-type alias Settings msg a =
-    { internalMsg : ( Msg a, Maybe a ) -> msg
-    , optionType : String
-    , optionLabel : a -> String
-    , optionDescription : a -> String
-    , searchFn : SearchUnion a
-    , debounceDuration : Float
-    , characterSearchThreshold : Int
-    , closeOnSelect : Bool
+selectSettings : Select.Settings Msg Product
+selectSettings =
+    { internalMsg = \msg -> HandleSelectUpdate msg
+    , searchFn =
+        \searchText products ->
+            List.filter (\product -> String.contains (String.toLower searchText) (String.toLower product.name)) products
+    , optionType = "Product"
+    , optionLabel = \product -> product.name
+    , optionDescription = \product -> product.price
+    , optionsContainerMaxHeight = 300
```

Comment:
It has a max height because the container is set to overflow (i.e. scroll). Otherwise the select options just go off the page in the event there are a lot of them (and the page itself isn't scrollable). This can be an internal setting, but I figured it wasn't a huge deal to give the user the ability to customize the size they want their options container to be.

Reply:
As additional context, the scroll container height and option height need to be specified to facilitate scrolling on down or up arrow press.

Reply:
Ah I didn't read the s at the end of options I thought this is for a single option... That's why I would like to not deal with this by default. I think it is nice that you can change these things but it just makes it so much more complicated. I really think we should introduce `AdvancedSettings` for those.